stdlib/optparse.pyi

```diff
@@ -209,18 +209,19 @@ class OptionGroup(OptionContainer):
     def set_title(self, title: str) -> None: ...
 
 class Values:
-    def __init__(self, defaults: Mapping[str, Incomplete] | None = None) -> None: ...
-    def _update(self, dict: Mapping[str, Incomplete], mode) -> None: ...
-    def _update_careful(self, dict: Mapping[str, Incomplete]) -> None: ...
-    def _update_loose(self, dict: Mapping[str, Incomplete]) -> None: ...
-    def ensure_value(self, attr: str, value): ...
-    def read_file(self, filename: str, mode: str = "careful") -> None: ...
-    def read_module(self, modname: str, mode: str = "careful") -> None: ...
+    def __init__(self, defaults: Mapping[str, object] | None = None) -> None: ...
+    def _update(self, dict: Mapping[str, object], mode: Literal["careful", "loose"]) -> None: ...
+    def _update_careful(self, dict: Mapping[str, object]) -> None: ...
+    def _update_loose(self, dict: Mapping[str, object]) -> None: ...
+    def ensure_value(self, attr: str, value: object) -> Any: ...  # return type cannot be known statically
+    def read_file(self, filename: str, mode: Literal["careful", "loose"] = "careful") -> None: ...
+    def read_module(self, modname: str, mode: Literal["careful", "loose"] = "careful") -> None: ...
     __hash__: ClassVar[None]  # type: ignore[assignment]
     # __getattr__ doesn't exist, but anything passed as a default to __init__
     # is set on the instance.
-    def __getattr__(self, name: str): ...
-    def __setattr__(self, name: str, value, /) -> None: ...
+    def __getattr__(self, name: str) -> Any: ...
+    # TODO mypy infers -> object for __getattr__ if __setattr__ has `value: object`
+    def __setattr__(self, name: str, value: Any, /) -> None: ...
```

Comment:
```diff
+ src/pip/_internal/commands/install.py:317: error: Argument 1 to "exists" has incompatible type "object"; expected "int | str | bytes | PathLike[str] | PathLike[bytes]"  [arg-type]
+ src/pip/_internal/commands/install.py:318: error: Argument 1 to "isdir" has incompatible type "object"; expected "int | str | bytes | PathLike[str] | PathLike[bytes]"  [arg-type]
```
Basically this behavior if `value` is annotated with  `object`:
```python
import optparse
import os

parser = optparse.OptionParser()
parser.add_option("--target-dir", dest="target_dir", help="Target directory")
values, rest = parser.parse_args(["--target-dir", "/tmp"])
reveal_type(values.target_dir)  # mypy: Any, pyright: Any
values.target_dir = os.path.abspath(values.target_dir)
reveal_type(values.target_dir)  # mypy: builtins.object, pyright: Any
```

I think this is a mypy bug because `__getattr__` returns `Any`. Let me know if you think this is expected or I should open a mypy issue instead.